nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/hook/NiFiAtlasHook.java

```diff
@@ -55,14 +55,23 @@ public void addMessage(HookNotificationMessage message) {
     }
 
     public void commitMessages() {
-        final NotificationSender notificationSender = new NotificationSender();
+        final NotificationSender notificationSender = createNotificationSender();
         notificationSender.setAtlasClient(atlasClient);
         notificationSender.send(messages, this::notifyEntities);
+        messages.clear();
```

Comment:
The fact that the `send` method accepts a function makes me think that the send happens asynchronously.  Are you sure that `send` is a blocking method?  If not, clearing the messages will cause a ConcurrentModificationException

Reply:
Good point, I've checked it now: the items from list are streamed / collected into other data structures at the beginning of send(), so it shouldn't be a problem and the list itself never used later.
Though, just to be on the safe side, I'll change it to create a new list after every commit.